packages/@aws-cdk/aws-iam/test/policy-document.test.ts

```diff
@@ -572,4 +591,197 @@ describe('IAM polocy document', () => {
 
     expect(stack.resolve(doc1)).toEqual(stack.resolve(doc2));
   });
+
+  describe('from JSON', () => {
+    test('no principal', () => {
+      const stack = new Stack();
+
+      const s = new PolicyStatement();
+      s.addActions('service:action1', 'service:action2');
+      s.addAllResources();
+      s.addCondition('key', { equals: 'value' });
+
+      const doc1 = new PolicyDocument();
+      doc1.addStatements(s);
+
+      const doc2 = PolicyDocument.fromJSON(doc1.toJSON());
+
+      expect(stack.resolve(doc2)).toEqual(stack.resolve(doc1));
+
+    });
+
+    test('from JSON arnPrincipal', () => {
+      const stack = new Stack();
+
+      const s = new PolicyStatement();
+      s.addActions('service:action1', 'service:action2');
+      s.addAllResources();
+      s.addArnPrincipal('somearn');
+      s.addCondition('key', { equals: 'value' });
+
+      const doc1 = new PolicyDocument();
+      doc1.addStatements(s);
+
+      const doc2 = PolicyDocument.fromJSON(doc1.toJSON());
+
+      expect(stack.resolve(doc2)).toEqual(stack.resolve(doc1));
+
+    });
+
+    test('from JSON anyPrincipal', () => {
```

Comment:
The "from JSON" prefix is not needed since all of these tests are within the group